x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/planner/LocalExecutionPlanner.java

```diff
@@ -364,9 +366,11 @@ private PhysicalOperation planExchangeSink(ExchangeSinkExec exchangeSink, LocalE
                     for (int i = 0, s = output.size(); i < s; i++) {
                         var out = output.get(i);
                         if (out.dataType() == DataTypes.BOOLEAN) {
+                            boolBlock.incRef();
```

Comment:
Should we create a new Block every time here? It's simpler and the cost is trivial (the number of blocks and the cost of ConstantBooleanVector should be small).

Reply:
Yep, that's even simpler.